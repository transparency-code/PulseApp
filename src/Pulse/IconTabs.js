//https://material-ui.com/components/tabs/
import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import ContactSupportOutlinedIcon from "@material-ui/icons/ContactSupportOutlined";
import ReceiptOutlinedIcon from '@material-ui/icons/ReceiptOutlined';
import QuestionAnswerOutlinedIcon from '@material-ui/icons/QuestionAnswerOutlined';
import GroupWorkOutlinedIcon from '@material-ui/icons/GroupWorkOutlined';
import AttachFileOutlinedIcon from '@material-ui/icons/AttachFileOutlined';

//https://css-tricks.com/almanac/properties/f/flex-grow/
//if all items have flex-grow set to 1, every child will set to an equal size inside the container
//https://gedd.ski/post/the-difference-between-width-and-flex-basis/
//https://material-ui.com/system/sizing/
//// Numbers are converted to pixel values.

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    maxWidth: 1000
  }
});

export default function IconTabs({companyUser=true}) {

    // console.log(companyUser)
  const classes = useStyles();
  const [value, setValue] = React.useState(0);

  //event _proto_ is a class
  //newVlaue is 0,1,2,3
  const handleChange = (event, newValue) => {
    //   console.log(event)
    setValue(newValue);
  };


  return (
    <Paper square className={classes.root}>
      <Tabs
        value={value}
        onChange={handleChange}
        variant="fullWidth"
        indicatorColor="primary"
        textColor="primary"
        aria-label="icon tabs example"
      >
        <Tab icon={<ContactSupportOutlinedIcon />} aria-label="request" />
        <Tab icon={<ReceiptOutlinedIcon />} aria-label="response" />
        <Tab icon={<QuestionAnswerOutlinedIcon />} aria-label="chat" />
        {companyUser && <Tab icon={<GroupWorkOutlinedIcon />} aria-label="team" />}
        <Tab icon={<AttachFileOutlinedIcon />} aria-label="files" />
      </Tabs>
    </Paper>
  );
}
