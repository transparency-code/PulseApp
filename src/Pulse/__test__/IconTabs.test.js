import React from 'react';
import IconTabs from '../IconTabs.js'
import renderer from 'react-test-renderer';

it('team is not rendered when companyUser is not set', () => {
    const tree = renderer
      .create(<IconTabs />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('team is  rendered when companyUser is set', () => {
    const tree = renderer
      .create(<IconTabs companyUser={true}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });