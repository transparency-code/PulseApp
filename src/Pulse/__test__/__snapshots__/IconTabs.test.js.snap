// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`team is  rendered when companyUser is set 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1"
>
  <div
    aria-label="icon tabs example"
    className="MuiTabs-root"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-label="request"
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected MuiTab-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 23.59v-3.6c-5.01-.26-9-4.42-9-9.49C2 5.26 6.26 1 11.5 1S21 5.26 21 10.5c0 4.95-3.44 9.93-8.57 12.4l-1.43.69zM11.5 3C7.36 3 4 6.36 4 10.5S7.36 18 11.5 18H13v2.3c3.64-2.3 6-6.08 6-9.8C19 6.36 15.64 3 11.5 3zm-1 11.5h2v2h-2zm2-1.5h-2c0-3.25 3-3 3-5 0-1.1-.9-2-2-2s-2 .9-2 2h-2c0-2.21 1.79-4 4-4s4 1.79 4 4c0 2.5-3 2.75-3 5z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="response"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.5 3.5L18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2 4.5 3.5 3 2v20l1.5-1.5L6 22l1.5-1.5L9 22l1.5-1.5L12 22l1.5-1.5L15 22l1.5-1.5L18 22l1.5-1.5L21 22V2l-1.5 1.5zM19 19.09H5V4.91h14v14.18zM6 15h12v2H6zm0-4h12v2H6zm0-4h12v2H6z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="chat"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 4v7H5.17l-.59.59-.58.58V4h11m1-2H3c-.55 0-1 .45-1 1v14l4-4h10c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1zm5 4h-2v9H6v2c0 .55.45 1 1 1h11l4 4V7c0-.55-.45-1-1-1z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="team"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
              <circle
                cx="8"
                cy="14"
                r="2"
              />
              <circle
                cx="12"
                cy="8"
                r="2"
              />
              <circle
                cx="16"
                cy="14"
                r="2"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="files"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        className="PrivateTabIndicator-root-63 PrivateTabIndicator-colorPrimary-64 MuiTabs-indicator"
        style={
          Object {
            "left": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`team is not rendered when companyUser is not set 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1"
>
  <div
    aria-label="icon tabs example"
    className="MuiTabs-root"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-label="request"
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected MuiTab-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 23.59v-3.6c-5.01-.26-9-4.42-9-9.49C2 5.26 6.26 1 11.5 1S21 5.26 21 10.5c0 4.95-3.44 9.93-8.57 12.4l-1.43.69zM11.5 3C7.36 3 4 6.36 4 10.5S7.36 18 11.5 18H13v2.3c3.64-2.3 6-6.08 6-9.8C19 6.36 15.64 3 11.5 3zm-1 11.5h2v2h-2zm2-1.5h-2c0-3.25 3-3 3-5 0-1.1-.9-2-2-2s-2 .9-2 2h-2c0-2.21 1.79-4 4-4s4 1.79 4 4c0 2.5-3 2.75-3 5z"
              />
            </svg>
          </span>
          <span
            className="PrivateTabIndicator-root-63 PrivateTabIndicator-colorPrimary-64 MuiTabs-indicator"
            style={Object {}}
          />
        </button>
        <button
          aria-label="response"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.5 3.5L18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2 4.5 3.5 3 2v20l1.5-1.5L6 22l1.5-1.5L9 22l1.5-1.5L12 22l1.5-1.5L15 22l1.5-1.5L18 22l1.5-1.5L21 22V2l-1.5 1.5zM19 19.09H5V4.91h14v14.18zM6 15h12v2H6zm0-4h12v2H6zm0-4h12v2H6z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="chat"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 4v7H5.17l-.59.59-.58.58V4h11m1-2H3c-.55 0-1 .45-1 1v14l4-4h10c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1zm5 4h-2v9H6v2c0 .55.45 1 1 1h11l4 4V7c0-.55-.45-1-1-1z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="files"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
